wire-library/wire-grpc-client/src/jvmMain/kotlin/com/squareup/wire/GrpcClient.kt

```diff
@@ -61,21 +62,33 @@ actual class GrpcClient private constructor(
     return RealGrpcStreamingCall(this, method)
   }
 
+  fun newBuilder(): Builder {
+    return Builder()
+      .callFactory(client)
+      .baseUrl(baseUrl)
+      .minMessageToCompress(minMessageToCompress)
+  }
+
   internal fun newCall(method: GrpcMethod<*, *>, requestBody: GrpcRequestBody): Call {
     return client.newCall(GrpcRequestBuilder()
-        .url(baseUrl.resolve(method.path)!!)
-        .addHeader("te", "trailers")
-        .addHeader("grpc-trace-bin", "")
-        .addHeader("grpc-accept-encoding", "gzip")
-        .addHeader("grpc-encoding", "gzip")
-        .tag(GrpcMethod::class.java, method)
-        .method("POST", requestBody)
-        .build())
+      .url(baseUrl.resolve(method.path)!!)
+      .addHeader("te", "trailers")
+      .addHeader("grpc-trace-bin", "")
+      .addHeader("grpc-accept-encoding", "gzip")
+      .apply {
+        if (minMessageToCompress < Long.MAX_VALUE) {
+          addHeader("grpc-encoding", "gzip")
+        }
+      }
+      .tag(GrpcMethod::class.java, method)
+      .method("POST", requestBody)
+      .build())
   }
 
   class Builder {
     private var client: Call.Factory? = null
     private var baseUrl: GrpcHttpUrl? = null
+    private var minMessageToCompress: Long = 0L
```

Comment:
No `MAX_VALUE` ?

Reply:
I wanted to introduce the feature independent of changing our default behavior. For easier rollbacks!